homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,263 @@
+            "alerted": self.is_alerted,
+            "recording": self.is_recording,
+            "has_ptz": self.device.has_ptz,
+            "motion_detection_enabled": self.device.detector_active,
+            "alerts_enabled": self.device.alerts_active,
+        }
+
+    @property
+    def is_recording(self) -> bool:
+        """Return whether the monitor is recording."""
+        return self.device.recording
+
+    @property
+    def is_alerted(self) -> bool:
+        """Return whether the monitor has alerted."""
+        return self.device.alerted
+
+    @property
+    def is_detected(self) -> bool:
+        """Return whether the monitor has alerted."""
+        return self.device.detected
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.device.client.is_available
+
+    @property
+    def connected(self) -> bool:
+        """Return True if entity is connected."""
+        return self.device.connected
+
+    @property
+    def enabled(self) -> bool:
+        """Return True if entity is enabled."""
+        return self.device.online
+
+    @property
+    def supported_features(self) -> int:
+        """Return supported features."""
+        return SUPPORT_ON_OFF | SUPPORT_STREAM
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if on."""
+        return self.device.online
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        if self.enabled:
+            return "mdi:camcorder"
+        return "mdi:camcorder-off"
+
+    @property
+    def state(self):
+        """Return the camera state."""
+        if self.device.recording:
+            return STATE_RECORDING
+        if self.device.online:
+            return STATE_STREAMING
+        return STATE_IDLE
+
+    async def stream_source(self) -> str:
+        """Return the mp4 stream source."""
+        return self._stream_url
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this agent object."""
+        return self._unique_id
+
+    async def async_enable_alerts(self):
+        """Enable alerts."""
+        await self.device.alerts_on()
+
+    async def async_disable_alerts(self):
+        """Disable alerts."""
+        await self.device.alerts_off()
+
+    async def async_enable_motion_detection(self):
+        """Enable motion detection."""
+        await self.device.detector_on()
+
+    async def async_disable_motion_detection(self):
+        """Disable motion detection."""
+        await self.device.detector_off()
+
+    async def async_start_recording(self):
+        """Start recording."""
+        await self.device.record()
+
+    async def async_stop_recording(self):
+        """Stop recording."""
+        await self.device.record_stop()
+
+    async def async_turn_on(self):
+        """Enable the camera."""
+        await self.device.enable()
+
+    async def async_toggle(self):
```

Comment:
Where is the toggle method used?

Reply:
It's left over functionality that has been removed during review